src/privileged/notificationBar/api.js

```diff
@@ -0,0 +1,155 @@
+"use strict";
+
+/* global ExtensionAPI */
+
+ChromeUtils.import("resource://gre/modules/Console.jsm");
+ChromeUtils.import("resource://gre/modules/Services.jsm");
+ChromeUtils.import("resource://gre/modules/XPCOMUtils.jsm");
+ChromeUtils.import("resource://gre/modules/ExtensionCommon.jsm");
+ChromeUtils.import("resource://gre/modules/ExtensionUtils.jsm");
+
+/* eslint-disable-next-line */
+var {EventManager, EventEmitter} = ExtensionCommon;
+/* eslint-disable-next-line no-var */
+var {Management: {global: {tabTracker}}} = ChromeUtils.import("resource://gre/modules/Extension.jsm", {});
+
+// eslint-disable-next-line no-undef
+XPCOMUtils.defineLazyModuleGetter(
+  this,
+  "BrowserWindowTracker",
+  "resource:///modules/BrowserWindowTracker.jsm",
+);
+
+/** Return most recent NON-PRIVATE browser window, so that we can
+ * manipulate chrome elements on it.
+ */
+function getMostRecentBrowserWindow() {
+  return BrowserWindowTracker.getTopWindow({
+    private: false,
+    allowPopups: false,
+  });
+}
+
+/** Display instrumented 'notification bar' explaining the feature to the user
+ *
+ *   Telemetry Probes:
+ *
+ *   - {event: survey-shown}
+ *
+ *   - {event: page-broken}
+ *
+ *   - {event: page-not-broken}
+ *
+ *    Note:  Bar WILL NOT SHOW if the only window open is a private window.
+ *
+ *    Note:  Handling of 'x' is not implemented.  For more complete implementation:
+ *
+ *      https://github.com/gregglind/57-perception-shield-study/blob/680124a/addon/lib/Feature.jsm#L148-L152
+ *
+ */
+class NotificationBarEventEmitter extends EventEmitter {
+  emitShow(variationName) {
```

Comment:
So this function spawns a notification which can emit either "page-broken" or "page-not-broken" right?
Is there any reason the UI code is packaged into this EventEmitter class? It's a bit confusing to read, I would have expected this class to simply implement two emit functions for the two events, and the UI code could just go directly in show() and call this class's functions.

Reply:
I'm not 100% certain how that should work, do you have functional/safety issues with this or just readability? I think it's quite easy to understand like this. If we're just talking about readability I'd prefer to avoid completely refactoring this module.